test/e2e/app_suffix_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/k14s/kapp/pkg/kapp/app"
+	"github.com/stretchr/testify/require"
+)
+
+var yaml1 string = `
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis-primary
+spec:
+  ports:
+  - port: 6380
+    targetPort: 6380
+  selector:
+    app: redis
+    tier: backend
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-config
+data:
+  key: value
+`
+
+var yaml2 string = `
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-config
+data:
+  key: value2
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-config2
+data:
+  key: value
+`
+
+func TestAppSuffix_AppExists_OldBehavior(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	kubectl := Kubectl{t, env.Namespace, logger}
+
+	name := "test-app-suffix-app-exists"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	os.Setenv("USE_OLD_CONFIGMAP_NAME", "True")
+
+	logger.Section("initial deploy", func() {
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+		NewPresentClusterResource("configmap", name, env.Namespace, kubectl)
+		NewMissingClusterResource(t, "configmap", name+app.AppSuffix, env.Namespace, kubectl)
+
+		cleanUp()
+	})
+
+	logger.Section("deploy with 1 delete, 1 update, 1 create", func() {
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+		NewPresentClusterResource("configmap", name, env.Namespace, kubectl)
+
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml2)})
+		NewPresentClusterResource("configmap", name, env.Namespace, kubectl)
+
+		cleanUp()
+	})
+
+	logger.Section("rename", func() {
```

Comment:
I think we can use the same app (from deploy section) for rename and delete. Although it's more of a personal taste, but the tests would definitely be a tiny bit faster if we use the same app.

Also, should we include tests for `inspect` command?

Reply:
gotcha, I will make the tests consistent with what already exists which follows the pattern you outlined. In my mind they were separate tests and a cascading side effect felt weird. But it's just a logger section, so I am on-board with that change :)